[Android] Loading of natives in static intializer makes it impossible to gracefully handle missing native libs

#### Issue details
Because GdxNativesLoader.load() is called in the static initializer of AndroidApplication, it's not possible to run any code before it. This means that if native libs are not present the app immediately crashes, with no way to check for missing natives and handle the situation more gracefully (such as by displaying an error popup and closing the app).

This is an issue because of Google's new app bundle functionality on Google Play. When app bundles are used, Google Play only distributes necessary resources to each device, including natives. This results in some nice space savings, but it also makes it very easy for someone to sideload the wrong APK and not get the natives they need.

I would think a better approach would be to load native libs in AndroidApplication.onCreate. This way code to check native libs could be run before calling super.onCreate. LibGDX could even perform the checking itself and display a standard error message, which a developer could then customize/override by manually checking earlier. This is much preferable to crashing as it won't affect an application's stability statistics on Google Play, and gives an opportunity to direct users to an APK which would work for them.

I'm not very familiar with the LibGDX codebase, so it's possible I'm missing something here, but if there is agreement that moving native initialization to AndroidApplication.onCreate is a good idea I'd be happy to submit a pull request.

#### Reproduction steps/code
Can be reproduced in any libgdx android project by simply removing native libraries and commenting out native dependencies in build.gradle

#### Version of LibGDX and/or relevant dependencies
Tested on 1.9.10 but should affect all versions.

#### Stacktrace
(Example stack trace from a user crash on my game: [Shattered Pixel Dungeon](https://github.com/00-Evan/shattered-pixel-dungeon))
```java
java.lang.ExceptionInInitializerError: 
  at java.lang.Class.newInstance (Class.java)
  at android.app.AppComponentFactory.instantiateActivity (AppComponentFactory.java:69)
  at androidx.core.app.CoreComponentFactory.instantiateActivity (CoreComponentFactory.java:43)
  at android.app.Instrumentation.newActivity (Instrumentation.java:1215)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3008)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3257)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1948)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
Caused by: com.badlogic.gdx.utils.GdxRuntimeException: 
  at com.badlogic.gdx.utils.SharedLibraryLoader.load (SharedLibraryLoader.java:125)
  at com.badlogic.gdx.utils.GdxNativesLoader.load (GdxNativesLoader.java:33)
  at com.badlogic.gdx.backends.android.AndroidApplication.<clinit> (AndroidApplication.java:60)
  at java.lang.Class.newInstance (Class.java)
  at android.app.AppComponentFactory.instantiateActivity (AppComponentFactory.java:69)
  at androidx.core.app.CoreComponentFactory.instantiateActivity (CoreComponentFactory.java:43)
  at android.app.Instrumentation.newActivity (Instrumentation.java:1215)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3008)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3257)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1948)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
Caused by: java.lang.UnsatisfiedLinkError: 
  at java.lang.Runtime.loadLibrary0 (Runtime.java:1012)
  at java.lang.System.loadLibrary (System.java:1669)
  at com.badlogic.gdx.utils.SharedLibraryLoader.load (SharedLibraryLoader.java:119)
  at com.badlogic.gdx.utils.GdxNativesLoader.load (GdxNativesLoader.java:33)
  at com.badlogic.gdx.backends.android.AndroidApplication.<clinit> (AndroidApplication.java:60)
  at java.lang.Class.newInstance (Class.java)
  at android.app.AppComponentFactory.instantiateActivity (AppComponentFactory.java:69)
  at androidx.core.app.CoreComponentFactory.instantiateActivity (CoreComponentFactory.java:43)
  at android.app.Instrumentation.newActivity (Instrumentation.java:1215)
  at android.app.ActivityThread.performLaunchActivity (ActivityThread.java:3008)
  at android.app.ActivityThread.handleLaunchActivity (ActivityThread.java:3257)
  at android.app.servertransaction.LaunchActivityItem.execute (LaunchActivityItem.java:78)
  at android.app.servertransaction.TransactionExecutor.executeCallbacks (TransactionExecutor.java:108)
  at android.app.servertransaction.TransactionExecutor.execute (TransactionExecutor.java:68)
  at android.app.ActivityThread$H.handleMessage (ActivityThread.java:1948)
  at android.os.Handler.dispatchMessage (Handler.java:106)
  at android.os.Looper.loop (Looper.java:214)
  at android.app.ActivityThread.main (ActivityThread.java:7050)
  at java.lang.reflect.Method.invoke (Method.java)
  at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run (RuntimeInit.java:493)
  at com.android.internal.os.ZygoteInit.main (ZygoteInit.java:965)
```

#### Please select the affected platforms
- [x] Android
- [ ] iOS (robovm)
- [ ] iOS (MOE)
- [ ] HTML/GWT
- [ ] Windows
- [ ] Linux
- [ ] MacOS

--- a/CHANGES
+++ b/CHANGES
@@ -1,4 +1,5 @@
 [1.10.1]
+- [BREAKING CHANGE] Android Moved natives loading out of static init block, see #5795
 - API Addition: ObjLoader now supports ambientColor, ambientTexture, transparencyTexture, specularTexture and shininessTexture
 - API Addition: PointSpriteParticleBatch blending is now configurable.
 - TOOLS Features: Blending mode and sort mode can be changed in Flame particle 3D editor.

--- a/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java
+++ b/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidApplication.java
@@ -40,9 +40,6 @@ import com.badlogic.gdx.utils.*;
  * 
  * @author mzechner */
 public class AndroidApplication extends Activity implements AndroidApplicationBase {
-	static {
-		GdxNativesLoader.load();
-	}
 
 	protected AndroidGraphics graphics;
 	protected AndroidInput input;
@@ -115,6 +112,7 @@ public class AndroidApplication extends Activity implements AndroidApplicationBa
 		if (this.getVersion() < MINIMUM_SDK) {
 			throw new GdxRuntimeException("LibGDX requires Android API Level " + MINIMUM_SDK + " or later.");
 		}
+		GdxNativesLoader.load();
 		setApplicationLogger(new AndroidApplicationLogger());
 		graphics = new AndroidGraphics(this, config, config.resolutionStrategy == null ? new FillResolutionStrategy()
 			: config.resolutionStrategy);

--- a/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidDaydream.java
+++ b/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidDaydream.java
@@ -45,9 +45,6 @@ import com.badlogic.gdx.utils.SnapshotArray;
  * @author mzechner */
 @TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR1)
 public class AndroidDaydream extends DreamService implements AndroidApplicationBase {
-	static {
-		GdxNativesLoader.load();
-	}
 
 	protected AndroidGraphics graphics;
 	protected AndroidInput input;
@@ -108,6 +105,7 @@ public class AndroidDaydream extends DreamService implements AndroidApplicationB
 	}
 
 	private void init (ApplicationListener listener, AndroidApplicationConfiguration config, boolean isForView) {
+		GdxNativesLoader.load();
 		setApplicationLogger(new AndroidApplicationLogger());
 		graphics = new AndroidGraphics(this, config, config.resolutionStrategy == null ? new FillResolutionStrategy()
 			: config.resolutionStrategy);

--- a/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidFragmentApplication.java
+++ b/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidFragmentApplication.java
@@ -33,10 +33,6 @@ public class AndroidFragmentApplication extends Fragment implements AndroidAppli
 		void exit ();
 	}
 
-	static {
-		GdxNativesLoader.load();
-	}
-
 	protected AndroidGraphics graphics;
 	protected AndroidInput input;
 	protected AndroidAudio audio;
@@ -132,6 +128,7 @@ public class AndroidFragmentApplication extends Fragment implements AndroidAppli
 		if (this.getVersion() < MINIMUM_SDK) {
 			throw new GdxRuntimeException("LibGDX requires Android API Level " + MINIMUM_SDK + " or later.");
 		}
+		GdxNativesLoader.load();
 		setApplicationLogger(new AndroidApplicationLogger());
 		graphics = new AndroidGraphics(this, config, config.resolutionStrategy == null ? new FillResolutionStrategy()
 			: config.resolutionStrategy);

--- a/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidLiveWallpaper.java
+++ b/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidLiveWallpaper.java
@@ -35,9 +35,6 @@ import com.badlogic.gdx.utils.*;
  * 
  * @author mzechner */
 public class AndroidLiveWallpaper implements AndroidApplicationBase {
-	static {
-		GdxNativesLoader.load();
-	}
 
 	protected AndroidLiveWallpaperService service;
 
@@ -64,6 +61,7 @@ public class AndroidLiveWallpaper implements AndroidApplicationBase {
 		if (this.getVersion() < MINIMUM_SDK) {
 			throw new GdxRuntimeException("LibGDX requires Android API Level " + MINIMUM_SDK + " or later.");
 		}
+		GdxNativesLoader.load();
 		setApplicationLogger(new AndroidApplicationLogger());
 		graphics = new AndroidGraphicsLiveWallpaper(this, config, config.resolutionStrategy == null ? new FillResolutionStrategy()
 			: config.resolutionStrategy);

--- a/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidLiveWallpaperService.java
+++ b/backends/gdx-backend-android/src/com/badlogic/gdx/backends/android/AndroidLiveWallpaperService.java
@@ -56,9 +56,6 @@ import com.badlogic.gdx.utils.GdxNativesLoader;
  * 
  * @author Jaroslaw Wisniewski <j.wisniewski@appsisle.com> */
 public abstract class AndroidLiveWallpaperService extends WallpaperService {
-	static {
-		GdxNativesLoader.load();
-	}
 
 	static final String TAG = "WallpaperService";
 	static boolean DEBUG = false; // TODO remember to disable this

--- a/gdx/src/com/badlogic/gdx/utils/GdxNativesLoader.java
+++ b/gdx/src/com/badlogic/gdx/utils/GdxNativesLoader.java
@@ -24,10 +24,10 @@ public class GdxNativesLoader {
 	/** Loads the libgdx native libraries if they have not already been loaded. */
 	static public synchronized void load () {
 		if (nativesLoaded) return;
-		nativesLoaded = true;
 
 		if (disableNativesLoading) return;
 
 		new SharedLibraryLoader().load("gdx");
+		nativesLoaded = true;
 	}
 }

